laborator/lab-03/README.md

```diff
@@ -0,0 +1,425 @@
+{
+    int err;
+    int x, y;
+ 
+    /* >>try<< instructions */
+    err = init_communication_with_mouse();
+    if (err)
+        goto error;
+ 
+    err = get_x_coord_from_mouse(&x);
+    if (err)
+        goto error_close_connection;
+ 
+    err = get_y_coord_from_mouse(&y);
+    if (err)
+        goto error_close_connection;
+ 
+    err = announce_upper_layers_of_mouse_movement(x, y);
+    if (err)
+        goto error_close_connection;
+ 
+    err = close_communication_with_mouse();
+    if (err)
+        goto error;
+ 
+    return 0;
+ 
+    /* >>catch<< instructions' exceptions */
+error_close_connection:
+    close_communication_with_mouse();
+error:
+    print_message("Failed to get data from mouse device. Error = %d", err);
+    return err;
+}
+```
+
+În varianta îmbunătățită, dacă apare o eroare, se face și o parte de curățenie:
+conexiunea cu mouse-ul va fi închisă, și apoi codul va continua cu tratarea generală a oricărei
+erori din program (afișarea unui mesaj de eroare).
+
+>**NOTE**: De ce abordează acest curs/laborator un astfel de subiect?<br>
+Când vom studia limbajul de asamblare vom observa că o bună parte din workflow seamănă cu un program format din goto-uri, 
+chiar dacă majoritatea instrucțiunilor unui limbaj de nivel înalt, chiar și precum C, sunt inexistente. Gândirea și
+programarea cu goto-uri ne pregătește pentru lucrul în limbajul de asamblare.
+
+>**WARNING**: În orice alt caz, această formă de programare ar trebui evitată pe cât posibil. 
+![goto.png]( https://imgs.xkcd.com/comics/goto.png)
+
+## De la C la program în rulare
+
+Etapele prin care trece un program din momentul în care este scris până când este rulat ca un proces sunt, in ordine:
+
+- preprocesare
+- compilare
+- asamblare
+- link editare
+
+În imaginea de mai jos sunt reprezentate si detaliate aceste etape:
+
+![phases-full.png](https://ocw.cs.pub.ro/courses/_media/iocla/laboratoare/phases-full.png?cache=)
+
+În etapa de **compilare** codul este tradus din cod de nivel înalt în limbaj de asamblare. Limbajul de asamblare este
+o formă human-readable a ce ajunge procesorul să execute efectiv. Dacă programele scrise în limbaje de nivel înalt ajung
+să fie portate ușor pentru procesoare diferite (arm, powerpc, x86, etc.), cele scrise în limbaj de asamblare sunt
+implementări specifice unei anumite arhitecturi. Limbaje de nivel înalt reprezintă o formă mai abstractă de rezolvare a
+unei probleme, din punctul de vedere al unui procesor, motiv pentru care și acestea trebuie traduse în limbaj de asamblare
+în cele din urmă, pentru a se putea ajunge la un binar care poate fi rulat. Mai multe detalii în laboratorul următor.
+
+Majoritatea compilatoarelor oferă opțiunea de a genera și un fișier cu programul scris în limbaj de asamblare.
+
+>**NOTE**: În cazul compilatorului **gcc** este de ajuns să adăugați flag-ul **-S** și vă va genera un
+fișier ***.s** cu codul aferent. În arhiva de **TODO** aveți un exemplu de trecere a unui program
```

Comment:
Din același motiv precum mai sus, ar fi preferabil să fie puse cu backtick-uri elementele evidențiate, în loc de bold. Mai ales cele care țin de secvențe de cod / comenzi în terminal / opțiuni ale comenzilor.